robotpy_build/hooks.py

```diff
@@ -567,6 +567,23 @@ def _function_hook(self, fn, data: FunctionData, internal: bool = False):
                     f"{fn['name']}: cannot specify ignore_pure for function that isn't pure"
                 )
 
+            if data.trampoline_cpp_code or data.virtual_xform:
+                if data.trampoline_cpp_code and data.virtual_xform:
+                    raise ValueError(
+                        f"{fn['name']}: cannot specify trampoline_cpp_code and virtual_xform"
```

Comment:
Can we put this data validation in a Pydantic validator method instead? https://pydantic-docs.helpmanual.io/usage/validators/